src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Base64Decoder.cs

```diff
@@ -558,40 +591,44 @@ private static OperationStatus DecodeWithWhiteSpaceBlockwise(ReadOnlySpan<byte>
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private static int GetPaddingCount(ref byte ptrToLastElement)
+        private static int GetPaddingCount<TBase64Decoder>(ref byte ptrToLastElement)
+            where TBase64Decoder : IBase64Decoder<byte>
         {
             int padding = 0;
 
-            if (ptrToLastElement == EncodingPad) padding++;
-            if (Unsafe.Subtract(ref ptrToLastElement, 1) == EncodingPad) padding++;
+            if (TBase64Decoder.IsValidPadding(ptrToLastElement))
+            {
+                padding++;
+            }
+
+            if (TBase64Decoder.IsValidPadding(Unsafe.Subtract(ref ptrToLastElement, 1)))
```

Comment:
What guarantees this is in bounds?

Reply:
This is only called with a buffer that has size of 4 
https://github.com/dotnet/runtime/blob/e617b8f47390727e6c211797051eac63a70881f5/src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Base64Decoder.cs#L498-L499
https://github.com/dotnet/runtime/blob/e617b8f47390727e6c211797051eac63a70881f5/src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Base64Decoder.cs#L545